parser/block.go

```diff
@@ -237,8 +237,10 @@ func (p *Parser) block(data []byte) {
 		// or
 		// ______
 		if p.isHRule(data) {
-			p.addBlock(&ast.HorizontalRule{})
 			i := skipUntilChar(data, 0, '\n')
+			hr := ast.HorizontalRule{}
+			hr.Literal = bytes.Trim(data[:i], " \n")
```

Comment:
this is probably overkill, your custom hook can do this.

should this be documented somewhere?

the change itself LGTM